### ISSUE TITLE
Loading pretrained dally, mismatch between checkpoint and initialized model

### ISSUE THREAD

**Issue by TheodoreGalanos** [lucidrains/DALLE-pytorch#89]
Hello!

So I have a partly trained dalle model that I wanted to test but I can't seem to load it. 

My vae params were:

```
{'image_size': 256,
 'num_layers': 3,
 'num_tokens': 2048,
 'codebook_dim': 512,
 'hidden_dim': 256,
 'num_resnet_blocks': 2}
```

My dalle params were:

```
'dim': 512,
'vae': vae,
'num_text_tokens': 49408,
'text_seq_len': 256,
'depth': 16,
'heads': 16,
'dim_head': 64,
'attn_dropout': 0.1,
'ff_dropout': 0.1,
'reversible': True,
'attn_types': ('full', 'axial_row', 'axial_col', 'conv_like')}
```

When I initialize both models and try to load my state_dict I get an error in the (I believe axial attention):

```
RuntimeError: Error(s) in loading state_dict for DALLE:
	size mismatch for image_pos_emb.weights_0: copying a param with shape torch.Size([1, 32, 1, 512]) from checkpoint, the shape in current model is torch.Size([1, 256, 1, 512]).
	size mismatch for image_pos_emb.weights_1: copying a param with shape torch.Size([1, 1, 32, 512]) from checkpoint, the shape in current model is torch.Size([1, 1, 256, 512]).
```

checking manually confirms that the initialized dalle model has a mismatch with the pretrained one.

image_pos_emb.weights_0 has a dimension of torch.Size([1, 1, 256, 512]) for the initialized model
image_pos_emb.weights_0 has a dimension of torch.Size([1, 1, 32, 512]) for the pretrained model

Wonder if I'm doing something wrong here? It's quite possible I've totally missed something!

**Comment by lucidrains** [comment]
@TheodoreGalanos Oh no, but that model you trained was prior to a bug fix to the positional encoding 🙏 You may have to just retrain it :(

**Comment by TheodoreGalanos** [comment]
Oh no! Oh well, it's okay I guess just a night's work. 

Can I ask, is it possible at all to hack my way into it? Maybe change the dimensions of that layer? Or is the whole model borked? I just want to test some outputs. 

Also, I pulled the repo about 24h ago. Do I just rerun it and it works or I need to pick a specific branch?

**Comment by lucidrains** [comment]
@TheodoreGalanos the fix was in this commit https://github.com/lucidrains/DALLE-pytorch/commit/01062110ac205337b1e076766b6cf67509d00059 , so you can try installing 0.1.25 `pip install dalle-pytorch==0.1.25` and see if that helps!

**Comment by TheodoreGalanos** [comment]
I just tested the new model I'm running (without convolutional attention) and it gives me the same error. I will try installing the version you mention (I simply run the bare command just now).

**Comment by TheodoreGalanos** [comment]
> 
> 
> @TheodoreGalanos the fix was in this commit [0106211](https://github.com/lucidrains/DALLE-pytorch/commit/01062110ac205337b1e076766b6cf67509d00059) , so you can try installing 0.1.25 `pip install dalle-pytorch==0.1.25` and see if that helps!

Sorry just a noob question, the current version I had installed was 0.3.4. Would that supersed the fix?

**Comment by TheodoreGalanos** [comment]
Nevermind, turns out the mismatch was in my local computer. All fixed after updated the dalle-pytorch package!
